test/src/graph/nodes/FilterGroundPointsNodeTest.cpp

```diff
@@ -0,0 +1,91 @@
+	rgl_node_t filterGroundNode;
+
+	FilterGroundPointsNodeTest() { filterGroundNode = nullptr; }
+};
+
+TEST_F(FilterGroundPointsNodeTest, invalid_argument_node)
+{
+	EXPECT_RGL_INVALID_ARGUMENT(rgl_node_points_filter_ground(nullptr, RGL_AXIS_Y, 0.0f), "node != nullptr");
+	EXPECT_RGL_INVALID_ARGUMENT(rgl_node_points_filter_ground(&filterGroundNode, RGL_AXIS_Y, -1.0f), "ground_angle_threshold >= 0");
+}
+
+TEST_F(FilterGroundPointsNodeTest, valid_argument_node)
+{
+	EXPECT_RGL_SUCCESS(rgl_node_points_filter_ground(&filterGroundNode, RGL_AXIS_Y, 1.0f));
+}
+
+TEST_F(FilterGroundPointsNodeTest, floor_test)
+{
+	// Scene
+	rgl_entity_t wall = makeEntity(makeCubeMesh());
+	rgl_mat3x4f wallPose = Mat3x4f::TRS(WALL_POS , Vec3f(0, 0, 0), WALL_DIMS).toRGL();
+	EXPECT_RGL_SUCCESS(rgl_entity_set_pose(wall, &wallPose));
+
+	// Rays
+	std::vector<rgl_mat3x4f> rays = makeLidar3dRays(360, 180, 0.36, 0.18);
+
+	// Graph
+	rgl_node_t useRaysNode = nullptr;
+	rgl_node_t transformRaysNode = nullptr;
+	rgl_node_t raytraceNode = nullptr;
+	rgl_node_t yieldNode = nullptr;
+
+	// Prepare graph without filtering
+	EXPECT_RGL_SUCCESS(rgl_node_rays_from_mat3x4f(&useRaysNode, rays.data(), rays.size()));
+	EXPECT_RGL_SUCCESS(rgl_node_rays_transform(&transformRaysNode, &identityTestTransform));
+	EXPECT_RGL_SUCCESS(rgl_node_raytrace(&raytraceNode, nullptr));
+	EXPECT_RGL_SUCCESS(rgl_node_points_yield(&yieldNode, basicFields.data(), basicFields.size()));
+
+	EXPECT_RGL_SUCCESS(rgl_graph_node_add_child(useRaysNode, transformRaysNode));
+	EXPECT_RGL_SUCCESS(rgl_graph_node_add_child(transformRaysNode, raytraceNode));
+	EXPECT_RGL_SUCCESS(rgl_graph_node_add_child(raytraceNode, yieldNode));
+
+	EXPECT_RGL_SUCCESS(rgl_graph_run(raytraceNode));
+	TestPointCloud outputPointCloud = TestPointCloud::createFromNode(yieldNode, basicFields);
+	auto fullCloudSize = outputPointCloud.getPointCount();
+
+	// Prepare graph with non-hit filtering
+	rgl_node_t compactNonHitNode = nullptr;
+	EXPECT_RGL_SUCCESS(rgl_graph_node_remove_child(raytraceNode, yieldNode));
+	EXPECT_RGL_SUCCESS(rgl_node_points_compact(&compactNonHitNode));
+	EXPECT_RGL_SUCCESS(rgl_graph_node_add_child(raytraceNode, compactNonHitNode));
+	EXPECT_RGL_SUCCESS(rgl_graph_node_add_child(compactNonHitNode, yieldNode));
+
+	EXPECT_RGL_SUCCESS(rgl_graph_run(raytraceNode));
+	TestPointCloud hitPointCloud = TestPointCloud::createFromNode(yieldNode, basicFields);
+	auto hitCloudSize = hitPointCloud.getPointCount();
+	// Compacted cloud should be smaller and contain only hit points
+	EXPECT_LT(hitCloudSize, fullCloudSize);
+
+	// Prepare graph with ground filtering
+	rgl_node_t groundFilterNode = nullptr;
+	rgl_node_t compactGroundNode = nullptr;
+	EXPECT_RGL_SUCCESS(rgl_graph_node_remove_child(compactNonHitNode, yieldNode));
+	//Add ground filtering with ground compacting
+	EXPECT_RGL_SUCCESS(rgl_node_points_filter_ground(&groundFilterNode, RGL_AXIS_Y, std::numbers::pi_v<float>/2.0f));
```

Comment:
BTW. I wonder why clang-format didn't fixed it. Not only in tests, but also this `int const*` would be probably fixed. Have you set up git hook?
```suggestion
	EXPECT_RGL_SUCCESS(rgl_node_points_filter_ground(&groundFilterNode, RGL_AXIS_Y, std::numbers::pi_v<float> / 2.0f));
```